### PR TITLE
deprecation of reloadable feature, http_skip_adding_content_length_to_upgrade

### LINKED ISSUE #25015: envoy_reloadable_features_http_skip_adding_content_length_to_upgrade deprecation
Your change #21990 (http: Do not add content-length: 0 header to UPGRADE requests) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by yeongjukang**
This PR is to solve #25015.
I had no idea that building and testing took so much time. Thank you so much for making Envoy.

Signed-off-by: Yeongju Kang <yeongjukang@outlook.com>

Commit Message: "deprecation of reloadable feature, http_skip_adding_content_length_to_upgrade"
Risk Level: low
Testing: //test/integration:redirect_integration_test doesn't fail on x64 mac. cannot reproduce CI's failure

Docs Changes: no
Release Notes: no
Platform Specific Features: no
Fixes #25015
Deprecated: envoy.reloadable_features.http_skip_adding_content_length_to_upgrade

### CODE REVIEW

**Review by yanavlasov — CHANGES_REQUESTED** [review]
/wait

**Comment by yeongjukang** [comment]
I'm really sorry for mass notification, rebased in wrong way for conflict.yaml.

File: source/common/http/conn_manager_utility.cc [hunk 1]
```diff
@@ -89,19 +89,15 @@ ConnectionManagerUtility::MutateRequestHeadersResult ConnectionManagerUtility::m
   // If this is a Upgrade request, do not remove the Connection and Upgrade headers,
   // as we forward them verbatim to the upstream hosts.
   if (Utility::isUpgrade(request_headers)) {
-    if (!Runtime::runtimeFeatureEnabled(
```
[the remark this answers is not in the document]
Comment (yeongjukang) [reply source/common/http/conn_manager_utility.cc:92]: Thanks. I removed.

File: test/integration/websocket_integration_test.cc [hunk 2]
```diff
@@ -538,8 +490,6 @@ TEST_P(WebsocketIntegrationTest,
     return;
   }
 
-  config_helper_.addRuntimeOverride(
```
Comment (yanavlasov) [inline test/integration/websocket_integration_test.cc:541]: You can remove this entire test, as it is now exactly the same as the one above it.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
Thank you for doing this work. This looks very good. Just one more little change, please.

/wait

File: test/integration/websocket_integration_test.cc (hunk 2 above)
Comment (yeongjukang) [reply test/integration/websocket_integration_test.cc:541]: Thanks, removed.

**Review by htuch — COMMENTED** [review]
/lgtm api

**Comment by yeongjukang** [comment]
@yanavlasov Sorry I didn't click resolvation button. Thank you for kind review.

**Comment by phlax** [comment]
cc @yanavlasov i think this is awaiting final review

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**